### ISSUE TITLE
Add integration tests using testcontainers ?

### ISSUE THREAD

**Issue by ieugen** [tamizhvendan/honeyeql#3]
Since this is a low level library that IMO, should be rock solid I think it is very important to have tests.

One way of running the tests against real servers is to use test containers. 
https://www.testcontainers.org/modules/databases/

**Comment by tamizhvendan** [comment]
Hi @ieugen,

I do understand the value of tests & how critical it is to stability. I do have a working test suite with close to 180+ test cases in my local machine and running it during every push via git hooks.  

Due to some bad design decisions that I took in the early days, it is not in good shape to run it via a CI server. 

It will take at least two full days to clean up the mess and set up the CI server. As I already constrained with my time and lot more features are yet to be supported, I am not currently focussing on it right now. 

Once things settled down, I will open-source the test suite and set up the CI.
